lua/aerial/backends/treesitter/init.lua

```diff
@@ -68,7 +52,33 @@ M.fetch_symbols_sync = function(bufnr)
   -- It is used to determine node parents for the tree structure.
   local stack = {}
   local ext = extensions[lang]
-  for match in query.iter_group_results(bufnr, "aerial", syntax_tree:root(), lang) do
+  ---@diagnostic disable-next-line: missing-parameter
+  for _, matches, metadata in query:iter_matches(syntax_tree:root(), bufnr) do
+    ---@note mimic nvim-treesitter's query.iter_group_results return values:
+    --       {
+    --         kind = "Method",
+    --         name = {
+    --           metadata = {
+    --             range = { 2, 11, 2, 20 }
+    --           },
+    --           node = <userdata 1>
+    --         },
+    --         type = {
+    --           node = <userdata 2>
+    --         }
+    --       }
+    --- Matches can overlap. The last match wins.
+    local match = vim.tbl_extend("force", {}, metadata)
+    for id, node in pairs(matches) do
+      -- iter_group_results prefers `#set!` metadata, keeping the behaviour
+      match = vim.tbl_extend("keep", match, {
+        [query.captures[id]] = {
+          metadata = metadata[id],
+          node = node,
+        },
+      })
```

Comment:
This is the only extra change I applied, as I discovered that previously used method prefers metadata over captures, and even if this is mostly invisible, the behaviour should remain preserved until it needs to change.